src/envelope/spacecharge/EnvelopeSpaceChargePush.cpp

```diff
@@ -67,5 +68,66 @@ namespace impactx::envelope::spacecharge
 
     }
 
+    void
+    space_charge3D_push (
+        RefPart const & AMREX_RESTRICT refpart,
+        Map6x6 & AMREX_RESTRICT cm,
+        amrex::ParticleReal bunch_charge,
+        amrex::ParticleReal ds
+    )
+    {
+        using namespace amrex::literals;
+
+        // skip calculations for trivial case
+        if (bunch_charge == 0_prt) { return; }
+
+        // initialize the linear transport map
+        Map6x6 R = Map6x6::Identity();
+
+        // physical constants and reference quantities
+        using ablastr::constant::SI::c;
+        using ablastr::constant::SI::ep0;
+        using ablastr::constant::math::pi;
+        amrex::ParticleReal const mass = refpart.mass;
+        amrex::ParticleReal const charge = refpart.charge;
+        amrex::ParticleReal const pt_ref = refpart.pt;
+        amrex::ParticleReal const betgam2 = std::pow(pt_ref, 2) - 1_prt;
+
+        // evaluate the 3D space charge intensity parameter from bunch charge
+        amrex::ParticleReal const rcN = std::abs(charge * bunch_charge) / (4_prt * pi * ep0 * mass * std::pow(c,2));
+        amrex::ParticleReal const coeff = ds * rcN / betgam2 * (1_prt/(5_prt * std::sqrt(5_prt)));
+
+        // set parameters for elliptic integrals
+        amrex::ParticleReal const errtol = 1.0e-3;
```

Comment:
Similar to MLMG options, we could in a follow-up PR make this a ParmParse option.